source/common/config/subscription_factory_impl.cc

```diff
@@ -146,6 +146,14 @@ SubscriptionPtr SubscriptionFactoryImpl::collectionSubscriptionFromUrl(
             callbacks, resource_decoder, stats, dispatcher_,
             Utility::configSourceInitialFetchTimeout(config), false, options);
       }
+      case envoy::config::core::v3::ApiConfigSource::AGGREGATED_GRPC: {
+        if (!Runtime::runtimeFeatureEnabled("envoy.reloadable_features.unified_mux")) {
+          throw EnvoyException("AGGREGATED_GRPC only supported for unified_mux");
```

Comment:
I'm probably missing something here, but why is unified mux needed here?

Reply:
Not sure I fully grok the strange behavior. If the cluster was updated, it should apply it IMHO.

If it doesn't require many changes then supporting non-unified might be beneficial, as one can easily change the flag and see that everything works.